File: src/Starfield2026.BgEditor/backend/src/lib/ohana/Textures/PocketMonsters/PT.ts
import { BinaryReader } from '../../Core/BinaryReader.js';
import { PkmnContainer } from '../../Containers/PkmnContainer.js';
import { FileIO, formatType } from '../../Core/FileIO.js';
import type { OTexture } from './GfTexture.js';

/**
 * Loads all monster textures from a PT Pokemon container.
 * Ported from OhanaCli.Formats.Textures.PocketMonsters.PT (C#).
 */
export class PT {
  static load(input: BinaryReader): OTexture[] {
    const textures: OTexture[] = [];

    const container = PkmnContainer.load(input);
    for (let i = 0; i < container.content.length; i++) {
      const entry = container.content[i];
      if (!entry.data) continue;
      const file = FileIO.load(BinaryReader.fromBuffer(entry.data));
      if (file.type === formatType.model) {
        const modelGroup = file.data as { texture: OTexture[] };
        textures.push(...modelGroup.texture);
      }
    }

    return textures;
  }
}

File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Garc/Textures/PocketMonsters/PT.cs
using System.Collections.Generic;
using System.IO;

using MiniToolbox.Garc.Containers;

namespace MiniToolbox.Garc.Textures.PocketMonsters
{
    class PT
    {
        /// <summary>
        ///     Loads all monster textures on a PT Pokémon container.
        /// </summary>
        /// <param name="data">The data</param>
        /// <returns>The monster textures</returns>
        public static List<RenderBase.OTexture> load(Stream data)
        {
            List<RenderBase.OTexture> textures = new List<RenderBase.OTexture>();
            RenderBase.OModelGroup models = new RenderBase.OModelGroup();

            OContainer container = PkmnContainer.load(data);
            for (int i = 0; i < container.content.Count; i++)
            {
                FileIO.LoadedFile file = FileIO.load(new MemoryStream(container.content[i].data));
                if (file.type == FileIO.formatType.model) textures.AddRange(((RenderBase.OModelGroup)file.data).texture);
            }

            return textures;
        }
    }
}
